lib/IDE/Utils.cpp

```diff
@@ -1252,12 +1252,55 @@ Expr *swift::ide::getBase(ArrayRef<Expr *> ExprStack) {
   return Base;
 }
 
-bool swift::ide::isDynamicCall(Expr *Base, ValueDecl *D) {
-  auto TyD = D->getDeclContext()->getSelfNominalTypeDecl();
-  if (!TyD)
+static bool isDeclSyntacticallyOverridable(ValueDecl *D) {
+  auto *NTD = D->getDeclContext()->getSelfNominalTypeDecl();
+  if (!NTD)
     return false;
 
-  if (isa<StructDecl>(TyD) || isa<EnumDecl>(TyD) || D->isFinal())
+  if (!(isa<ClassDecl>(NTD) || isa<ProtocolDecl>(NTD)) || D->isFinal() ||
+      NTD->isFinal() || NTD->isActor())
+    return false;
```

Comment:
A little bit of a nitpick because I don’t like complicated `if` conditions too much: I would split this up into two `if` statements.

```suggestion
  if (!isa<ClassDecl>(NTD) && !isa<ProtocolDecl>(NTD)) {
    // Only classes and protocols support overloading by subtypes.
    return false;
  }
  if (D->isFinal() || NTD->isFinal() || NTD->isActor()) {
    // Final types and declarations aren't dynamic.
    return false;
  }
```

Reply:
Yeah that's fair. Will do :)